impress-org/givewp
# feature: Add rotating "ProTip" message to Donation table page

## Description

This PR introduces an enhancement aimed at providing users with additional information about our plugins add-ons, especially those who may not be aware of them. The primary feature is a rotating "ProTip" message that will be displayed exclusively on the Donations table. The message will randomly rotate among various available options. Each option includes UTM links that, when clicked, will open the respective add-on's documentation page in a new window.

To ensure a seamless user experience, the PR includes functionality to track and save the user's interaction with the options. This information will be stored in the database under the options table. If a user has already installed a specific add-on, it will be automatically removed from the list of options to display, preventing redundant or unnecessary messages from appearing.

### Options list includes: Addon - option _name
- Recurring Donations addon -  'givewp_donations_recurring_recommendation_dismissed'
- Fee Recovery addon - 'givewp_donations_fee_recovery_recommendation_dismissed' 
- Designated Funds - 'givewp_donations_designated_funds_recommendation_dismissed' 

## Affects

Donations Table Page

## Visuals

https://www.loom.com/share/6615c2522bd64c8d87422e64c5e6936d

## Testing Instructions

- Install and activate all 3 addons
- Remove each addon one at a time and visit the Donation Table Page to confirm that the "ProTip" message only displays if the addon is not active.
- Deactivate all the 3 addons and view the Donation Table Page while refreshing or changing pages multiple times to confirm that all 3 messages randomly populate one at a time.
- Remove one at a time and refresh or change pages multiple times to confirm the the removed addon message no long populates.
- Repeat previous step for all 3 addons


## Pre-review Checklist
